eras/conway/impl/src/Cardano/Ledger/Conway/Rules/GovCert.hs

```diff
@@ -159,9 +194,23 @@ conwayGovCertTransition = do
     ConwayResignCommitteeColdKey coldK -> do
       checkColdKeyHasNotResigned coldK vsCommitteeHotKeys
       pure $ vState {vsCommitteeHotKeys = Map.insert coldK Nothing vsCommitteeHotKeys}
-    ConwayUpdateDRep cred _mAnchor -> do
+    ConwayUpdateDRep cred mAnchor -> do
       Map.notMember cred vsDReps ?! ConwayDRepNotRegistered cred
-      pure vState -- TODO: update anchor
+      expiryEpoch <-
+        calcDRepExpiryEpoch
+          cgceCurrentSlot
+          (cgcePParams ^. ppDRepActivityL)
+      pure $
+        vState
+          & vsDRepsL
+            %~ Map.update
+              ( \x ->
+                  Just $
+                    x
+                      & drepExpiryL .~ expiryEpoch
+                      & drepAnchorL .~ mAnchor
+              )
+              cred
```

Comment:
Just for clarity, I think it would be good to extract this update function into a separate let binding
```suggestion
      let updateDRepState drepState =
            drepState & drepExpiryL .~ expiryEpoch
                      & drepAnchorL .~ mAnchor
      pure $ vState & vsDRepsL %~ Map.update (Just . updateDRepState) cred
```

Reply:
wow, fourmolu absolutely massacred this